packages/cli/test/integration/credentials.ee.test.ts

```diff
@@ -0,0 +1,270 @@
+import * as testDb from './shared/testDb';
+import { RESPONSE_ERROR_MESSAGES } from '../../src/constants';
+import { CredentialsEntity } from '../../src/databases/entities/CredentialsEntity';
+
+jest.mock('../../src/telemetry');
+
+let app: express.Application;
+let testDbName = '';
+let globalOwnerRole: Role;
+let globalMemberRole: Role;
+let credentialOwnerRole: Role;
+let saveCredential: SaveCredentialFunction;
+
+beforeAll(async () => {
+	app = await utils.initTestServer({
+		endpointGroups: ['credentials'],
+		applyAuth: true,
+	});
+	const initResult = await testDb.init();
+	testDbName = initResult.testDbName;
+
+	utils.initConfigFile();
+
+	globalOwnerRole = await testDb.getGlobalOwnerRole();
+	globalMemberRole = await testDb.getGlobalMemberRole();
+	credentialOwnerRole = await testDb.getCredentialOwnerRole();
+	saveCredential = affixRoleToSaveCredential(credentialOwnerRole);
+
+	utils.initTestLogger();
+	utils.initTestTelemetry();
+});
+
+beforeEach(async () => {
+	await testDb.truncate(['User', 'SharedCredentials', 'Credentials'], testDbName);
+	config.set('deployment.paid', true);
+});
+
+afterAll(async () => {
+	await testDb.terminate(testDbName);
+});
+
+// ----------------------------------------
+// dynamic router switching
+// ----------------------------------------
+
+test('dynamic router switching', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
+	const savedCredential = await saveCredential(credentialPayload(), { user: owner });
+
+	// free router
+	config.set('deployment.paid', false);
+
+	const shareResponse = agent
+		.post(`/credentials/${savedCredential.id}/share`)
+		.send({ shareeId: member.id });
+
+	const getResponse = agent.get(`/credentials/${savedCredential.id}`).send();
+
+	const [{ statusCode: shareStatus }, { statusCode: getStatus }] = await Promise.all([
+		shareResponse,
+		getResponse,
+	]);
+
+	expect(shareStatus).toBe(404);
+	expect(getStatus).toBe(200);
+
+	// EE router
+	config.set('deployment.paid', true);
+
+	const shareResponseEE = agent
+		.post(`/credentials/${savedCredential.id}/share`)
+		.send({ shareeId: member.id });
+
+	const getResponseEE = agent.get(`/credentials/${savedCredential.id}`).send();
+
+	const [{ statusCode: shareStatusEE }, { statusCode: getStatusEE }] = await Promise.all([
+		shareResponseEE,
+		getResponseEE,
+	]);
+
+	expect(shareStatusEE).toBe(200);
+	expect(getStatusEE).toBe(200);
+});
+
+// ----------------------------------------
+// share
+// ----------------------------------------
+
+test('POST /credentials/:id/share should share the credential', async () => {
+	const owner = await testDb.createUser({ globalRole: globalOwnerRole });
+	const member = await testDb.createUser({ globalRole: globalMemberRole });
+	const agent = utils.createAgent(app, { auth: true, user: owner });
```

Comment:
yes and no,

yes, it is boilerplate

and no because it describes how the test is setup. So if you for example now want to change naming, role or add properties for a test due to changes, then you need to copy from the wrapper or break it up instead of simply changing your part directly in the test